ci_scripts/ci/travis/run.sh

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env bash
+
+export CI_SITE_NAME="Travis"
+export TRAVIS_JOBNAME="macos-ninja"
+export CI_SOURCE_DIR=${TRAVIS_BUILD_DIR}
+export CI_ROOT_DIR="${CI_SOURCE_DIR}/.."
+export CI_BIN_DIR="${CI_ROOT_DIR}/${TRAVIS_JOBNAME}"
+# export COMMIT_RANGE="${TRAVIS_COMMIT_RANGE/.../ }"
+
+if [ -n "${TRAVIS_PULL_REQUEST_BRANCH}" ]
+then
+  export CI_BUILD_NAME="pr${TRAVIS_PULL_REQUEST}_${TRAVIS_PULL_REQUEST_BRANCH}_${TRAVIS_BUILD_NUMBER}_${TRAVIS_JOBNAME}"
+else
+  export CI_BUILD_NAME="${TRAVIS_BRANCH}_${TRAVIS_BUILD_NUMBER}_${TRAVIS_JOBNAME}"
+fi
+
+CTEST_SCRIPT=ci_scripts/ci/cmake/travis-macos-ninja.cmake
+
+CTEST=$(which ctest)
+
+# Where did HomeBrew install vtk?
+config_full_path=$(find /usr/local/Cellar | grep -i vtkconfig\\.cmake)
+
+if [ -z "${config_full_path}" ]; then
+    echo "Unable to find VTKConfig.cmake, please run 'brew install vtk' and try again"
+    exit 1
+fi
+
+export VTK_DIR=$(dirname "${config_full_path}")
+echo "Using VTK_DIR=${VTK_DIR}"
+
+export TMPDIR=/tmp
+
+echo "**********CTest Begin**********"
+${CTEST} --version
+
+# Define dashboard steps
+declare -a steps=("update" "configure" "build" "test")
```

Comment:
why run this in a loop? it seems to me to add complexity vs the usual approach

Reply:
Is your question about why it's done in steps?  Or just about the use of a bash loop instead of 4 separate invocations?